src/index.ts

```diff
@@ -367,14 +378,18 @@ export class ProjectConfig {
    * will be relative to root.
    */
   toJSON(): string {
-    const relative = (p: string | null | undefined) =>
-        p ? path.relative(this.root, p) : undefined;
     const obj = {
-      entrypoint: relative(this.entrypoint),
-      shell: relative(this.shell),
-      fragments: (this.fragments || []).map(relative),
-      sources: (this.sources || []).map(relative),
-      extraDependencies: (this.extraDependencies || []).map(relative),
+      entrypoint: globRelative(this.root, this.entrypoint),
+      shell: this.shell ? globRelative(this.root, this.shell) : undefined,
+      fragments: this.fragments.map((absolutePath) => {
```

Comment:
could be `(absolutePath) => globRelative(this.root, absolutePath);`

unless we don't like that syntax for some reason?